keanu-project/src/main/java/io/improbable/keanu/util/json/JsonSaver.java

```diff
@@ -0,0 +1,105 @@
+package io.improbable.keanu.util.json;
+
+import com.google.protobuf.util.JsonFormat;
+import io.improbable.keanu.KeanuSavedBayesNet;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.network.NetworkSaver;
+import io.improbable.keanu.network.ProtobufSaver;
+import io.improbable.keanu.vertices.ConstantVertex;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.bool.nonprobabilistic.ConstantBoolVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.intgr.IntegerVertex;
+import io.improbable.keanu.vertices.intgr.nonprobabilistic.ConstantIntegerVertex;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.Map;
+
+/**
+ * Utility class for outputting a network to a JSON file.
+ * JSON output file contains information about the network (vertices, their types and connections),
+ * network state (values for constant and observed vertices, if saveValues parameter is set to true),
+ * as well as metadata (string key - value pairs, if metadata is passed in).
+ *
+ * Usage:
+ * Create jsonSaver: JsonSaver jsonSaver = new JsonSaver(yourBayesianNetwork);
+ * or JsonSaver jsonSaver = new JsonSaver(yourBayesianNetwork, metadata);
+ * where metadata is a map between some keys and values in string format (for instance, "author": "Jane Doe").
+ * To output network to a JSON file: jsonSaver.save(outputStream, saveValues);
+ */
+public class JsonSaver implements NetworkSaver{
+
+    private final ProtobufSaver protobufSaver;
+
+    /**
+     * Sets up a new json saver for the given network.
+     * @param net network that will be saved
+     */
+    public JsonSaver(BayesianNetwork net) {
+        protobufSaver = new ProtobufSaver(net);
+    }
+
+    @Override
+    public void save(OutputStream output, boolean saveValues, Map<String, String> metadata) throws IOException {
+        KeanuSavedBayesNet.Model model = protobufSaver.getModel(saveValues, metadata);
+
+        Writer outputWriter = new OutputStreamWriter(output);
+        String jsonOutput = JsonFormat.printer().print(model);
+        outputWriter.write(jsonOutput);
+        outputWriter.close();
+    }
+
+    public void save(OutputStream output, boolean saveValues) throws IOException {
+        save(output, saveValues, null);
```

Comment:
This is the default implementation of the `NetworkSaver` interface. Could use that, if not this method should be annotated with `@Override`.